### PR TITLE
[DO NOT MERGE] Refactor JotForm answer format - remove value type (let client handle…

### PR DESCRIPTION
**Pull request by aoby**
… that), simplify, and add missing data such as options to the question summary

As it stands, this is incomplete and I expect it to fail tests. The survey results view client needs to be updated to accommodate the new format and convert certain `singleSelect` answers into numeric values based on index (work that used to be done in the server). @mehalshah is owning this part.

Future low-pri idea: Now that the logic is simpler, we could potentially add an improvement in the future to separate out validation and answer processing. Then we could add model validations and make the answer processing (once validated) more performant.

### CODE REVIEW

File: dashboard/app/models/teacher_feedback.rb [hunk 1]
```diff
@@ -6,14 +6,13 @@
 #  comment    :text(65535)
```
Comment (mehalshah) [inline dashboard/app/models/teacher_feedback.rb]: Is this change intentional?

File: dashboard/lib/pd/jot_form/matrix_question.rb [hunk 2]
```diff
@@ -49,15 +49,15 @@ def get_value(answer)
...
+        # Validate each answer and convert each key to sub_question_index
         answer.reject {|_, v| v.blank?}.map do |sub_question, sub_answer|
           sub_question_index = sub_questions.index(sub_question)
           raise "Unable to find sub-question '#{sub_question}' in matrix question #{id}" unless sub_question_index
 
-          sub_answer_index = options.index(sub_answer)
-          raise "Unable to find '#{sub_answer}' in the options for matrix question #{id}" unless sub_answer_index
+          raise "Unable to find '#{sub_answer}' in the options for matrix question #{id}" unless options.include? sub_answer
 
           # Return a 1-based value
```
Comment (mehalshah) [inline dashboard/lib/pd/jot_form/matrix_question.rb:59]: This comment no longer applies

File: dashboard/test/lib/pd/jot_form/form_questions_test.rb [hunk 3]
```diff
@@ -11,18 +11,21 @@ class FormQuestionsTest < ActiveSupport::TestCase
         @questions = [
           TextQuestion.new(
             id: 1,
+            order: 1,
```
Comment (aoby) [inline dashboard/test/lib/pd/jot_form/form_questions_test.rb:14]: I added `order` to each question here both to more accurately represent our real data (JotForm Qs always have an order), and also because for some reason the sort order changed without it and failed tests. I presume [sort_by](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/lib/pd/jot_form/form_questions.rb#L69) with all `nil` values has indeterminate results.

Comment (mehalshah) [reply dashboard/test/lib/pd/jot_form/form_questions_test.rb:14]: Thats perfect
